scrapper6pm/scrapper6pm/spiders/a6pm.py

```diff
@@ -0,0 +1,89 @@
+import json
+
+from scrapy.linkextractors import LinkExtractor
+from scrapy.spiders import CrawlSpider, Rule
+
+
+class A6pmSpider(CrawlSpider):
+    name = "6pm"
+    allowed_domains = ["www.6pm.com"]
+    start_urls = ["https://www.6pm.com/"]
+    data = {}
+
+    rules = (
+        Rule(LinkExtractor(restrict_css='[data-sub-nav="true"] a.fi-z')),
+        Rule(LinkExtractor(restrict_css='[id="searchPagination"] a'), follow=True),
+        Rule(LinkExtractor(restrict_css='[id="products"] a[itemprop="url"]'), callback="parse_item"),
+    )
+
+    def parse_item(self, response):
+        retailer_sku = self.extract_retailer_sku(response)
+        if retailer_sku in self.data:
+            return
+        item_dict = {
+            "retailer_sku": self.extract_retailer_sku(response),
+            "gender": self.extract_item_gender(response),
+            "brand": {
+                "name": response.css("div#breadcrumbs div a:last-child span::text").get(),
+                "logo": response.css('a[data-track-value="Brand-Logo"] img::attr(src)').get(),
+            },
+            "category": response.css("div#breadcrumbs div a::text").extract()[1:],
+            "url": response.url,
+            "name": response.css("span.Ws-z::text").get(),
+            "description": response.css("div[itemprop='description'] div.uV-z ul").get(),
+            "currency": response.css("span.fu-z.iu-z span:first-child span:first-child::attr(content)").get(),
+            "skus": self.extract_item_skus(response),
+        }
+        self.data[retailer_sku] = item_dict
```

Comment:
No need to store complete data. You can just store parsed retailer skus, as we have no concern with data.